### PR TITLE
File structure

### PR DESCRIPTION
**Pull request by RazvanDH**
Modified the file structure and some new mixins / functions.

### CODE REVIEW

**Comment by igoratron** [comment]
I am thinking we should `gitignore` the dist folder so that it doesn't get committed automatically and only build and commit it when we publish a new version/tag.

File: .gitignore [hunk 1]
```diff
@@ -1,6 +1,9 @@
+.DS_Store
```
Comment (jspc) [inline .gitignore]: This really ought to be in a [Global gitignore](https://help.github.com/articles/ignoring-files/#create-a-global-gitignore) given that it is very platform dependent

**Comment by sgerrand** [comment]
@RazvanDH: Would you please rebase `master` onto your branch to enable these changes to be merged.

File: .gitignore [hunk 2]
```diff
@@ -1,6 +1,9 @@
...
 .sass-cache
 
 bower_components
 node_modules
 
+npm-debug.log
+
```
Comment (jspc) [inline .gitignore]: Why the whitespace?

**Comment by jspc** [comment]
:+1: on ignoring `dist/`

**Pull request merged**